roles/openshift_health_checker/action_plugins/openshift_health_check.py

```diff
@@ -1,99 +1,124 @@
-            check = known_checks[check_name]
-
-            if not check.is_active():
-                r = dict(skipped=True, skipped_reason="Not active for this host")
-            elif check_name in user_disabled_checks:
-                r = dict(skipped=True, skipped_reason="Disabled by user request")
-            else:
-                try:
-                    r = check.run()
-                except OpenShiftCheckException as e:
-                    r = dict(
-                        failed=True,
-                        msg=str(e),
-                    )
-
-            check_results[check_name] = r
-
-            if r.get("failed", False):
-                result["failed"] = True
-                result["msg"] = "One or more checks failed"
+        for name in resolved_checks:
+            display.banner("CHECK [{} : {}]".format(name, task_vars["ansible_host"]))
+            check = known_checks[name]
+            check_results[name] = run_check(name, check, user_disabled_checks)
 
         result["changed"] = any(r.get("changed", False) for r in check_results.values())
+        result["failed"] = any(r.get("failed", False) for r in check_results.values())
+        if result["failed"]:
+            result["msg"] = "One or more checks failed"
         return result
 
     def load_known_checks(self, tmp, task_vars):
+        """Find all existing checks and return a mapping of names to instances."""
         load_checks()
 
         known_checks = {}
         for cls in OpenShiftCheck.subclasses():
-            check_name = cls.name
-            if check_name in known_checks:
-                other_cls = known_checks[check_name].__class__
-                raise OpenShiftCheckException(
-                    "non-unique check name '{}' in: '{}.{}' and '{}.{}'".format(
-                        check_name,
-                        cls.__module__, cls.__name__,
-                        other_cls.__module__, other_cls.__name__))
-            known_checks[check_name] = cls(execute_module=self._execute_module, tmp=tmp, task_vars=task_vars)
+            name = cls.name
+            if name in known_checks:
+                other_cls = known_checks[name].__class__
+                msg = "non-unique check name '{}' in: '{}' and '{}'".format(
+                    name,
+                    full_class_name(cls),
+                    full_class_name(other_cls),
+                )
+                raise OpenShiftCheckException(msg)
+            known_checks[name] = cls(execute_module=self._execute_module, tmp=tmp, task_vars=task_vars)
         return known_checks
 
 
+def list_known_checks(known_checks):
+    """Return text listing the existing checks and tags."""
+    # TODO: we could include a description of each check by taking it from a
+    # check class attribute (e.g., __doc__) when building the message below.
+    msg = (
+        'This playbook is meant to run health checks, but no checks were '
```

Comment:
I know what you mean, it looks bad if your terminal is thinner than the arbitrary point where we've placed newlines so it wraps anyway. I guess the real issue with that is on the check outputs since they're shown as indented blocks - tried to keep those short enough so that most people won't see them wrap. If we wanted to do that right we'd need to know the width of the terminal and have a real formatter on the output...

Just in general though I think for prose it makes sense to keep lines pretty short, like in a book. I may have a terminal that's 200 columns wide to look at logs and code and JSON, but it doesn't mean I want to read text at that width. I suppose ideally we'd have a formatter that broke on whitespace at the shorter of the terminal width or a comfortable-to-read width.

Well, I don't really want to start a typographical debate, just saying personally I would wrap it for legibility and consistency... it stuck out at me.